core/arch/instr_shared.c

```diff
@@ -1111,6 +1113,23 @@ instr_allocate_raw_bits(dcontext_t *dcontext, instr_t *instr, uint num_bytes)
 #endif
 }
 
+void
+instr_set_label_callback(instr_t *instr, instr_label_callback_t cb)
+{
+    CLIENT_ASSERT(instr_is_label(instr),
+                  "only set callback functions for label instructions");
+    CLIENT_ASSERT(instr->label_cb == NULL, "label callback function is already set");
```

Comment:
Add an assert that `!TEST(INSTR_RAW_BITS_ALLOCATED, instr->flags)` as a sanity check that we're not using both parts of the new union

Reply:
Done.